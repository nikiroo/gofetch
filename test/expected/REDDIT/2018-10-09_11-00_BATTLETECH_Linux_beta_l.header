0BATTLETECH Linux beta loading ten times faster than on Windows 10? Oh, the irony...	null/REDDIT/2018-10-09_11-00_BATTLETECH_Linux_beta_l		70
i  [linux_gaming] 2018-10-09_11-00
i
