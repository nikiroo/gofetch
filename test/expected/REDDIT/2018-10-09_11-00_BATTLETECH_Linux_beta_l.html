<!DOCTYPE html>
<html>
<head>
  <meta http-equiv='content-type' content='text/html; charset=utf-8'>
  <meta name='viewport' content='width=device-width, initial-scale=1.0'>
  <style type='text/css'>
    body { margin: 1em 15%; }
  </style>
</head>
<body>
<div class='story'>
	<h2>BATTLETECH Linux beta loading ten times faster than on Windows 10? Oh, the irony...</h2>
	<div class='details'>([linux_gaming] 2018-10-09_11-00)</div>
	<br/>
    <ul>
    <li>Reference: <a href=''>2018-10-09_11-00_BATTLETECH_Linux_beta_l</a></li>
        <li>News link: <a href='https://www.reddit.com/r/linux_gaming/comments/9ld2i1/battletech_linux_beta_loading_ten_times_faster/'>https://www.reddit.com/r/linux_gaming/comments/9ld2i1/battletech_linux_beta_loading_ten_times_faster/</a></li>
        <li>Source link: <a href='https://steamcommunity.com/app/637090/discussions/0/1693788384146305062/?ctp=13#c2747650363470418004'>https://steamcommunity.com/app/637090/discussions/0/1693788384146305062/?ctp=13#c2747650363470418004</a></li>
    </ul>
	<br/>
	<div class='content' style='text-align: justify'>
		ive always noticed that stuff loads much faster on Linux compared to windows, can someone explain why?
	</div>
<hr/>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>zee220</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>ive always noticed that stuff loads much faster on Linux compared to windows, can someone explain why?</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>pwgen-sy100</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Better filesystem's, better IO scheduling, better scheduling in general, less chances of some stupid ass av to clog up all the io is what I've found.</p></div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>breakbeats573</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>Unless you open calculator on Ubuntu. It's comical actually.</p></div>
      </div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>zebediah49</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>Also what appears to be more aggressive caching, and less bloat to burn through memory that would otherwise be cached.</p><p>I've note looked too carefully at it, but Windows machines usually appear to have a whole lot of free memory, and relatively little cached.</p><p>Meanwhile, my laptop has 10GB of "stuff" just floating around in memory in case I happen to want it.</p></div>
      </div>
    </div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>jhansonxi</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Yesterday I updated a Win10 laptop that had been sitting on a shelf for two years. I could have installed a dozens systems with *buntu using a online install from a mini boot image, with all major apps installed and updated, in the time it took Win10 to finish its updates, update reattempts, and dozen or so reboots. What's worse is that most of the Win10 updates were either for the OS or Office only. Any Linux package manager updates nearly everything and Steam handles most of the rest.</p></div>
    </div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>Batolemaeus</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>AV is the worst offender because it hooks into all I/O. But there's also significant overhead in NTFS when dealing with many small files. It's partially the cost of the much more involved security model on NT (and ACL processing). That's why compiling has an extreme performance penalty on Windows, since it reads and writes many small files.</p></div>
    </div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>aaronfranke</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>In addition to what's already been said, EXT4 doesn't fragment nearly as much as NTFS.</p></div>
    </div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>Craftkorb</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Linux is pretty good when it comes to utilizing (otherwise) free memory, using it to cache ("buffer") parts of your drives. Just try to do an reading-intensive I/O task, like find / -name stdio.h . The first may take a while, but then just rerun the command. It's much quicker, because Linux cached (in this case) the file structure the first time, and now can just quickly grab it from there again for the second time.</p><p>Want to see the caching of files (and not structures) in action? Find any medium sized file (about a gigabyte in size). Now, call md5sum path/to/the/file on it. Call it again afterwards. It should be much faster.</p><p>Don't have such a file handy? You can generate a "large.file" using dd if=/dev/zero of=large.file bs=4096 count=262144 . Do note that by doing this, Linux will already have (all or parts of) the file cached, the difference between both calls should be much smaller.</p><p>A much more file-centric system basically requires a good I/O cache to get good performance. Crucial for slow spinning drives (or worse, magnetic drives), but still really important for SSDs.</p></div>
    </div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>robot_rover</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>It’s because windows search and windows compatibility telemetry are currently using 100% or ur hard disk time</p></div>
    </div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>JackCourtney</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>Not a game, but GIMP loads so much faster on linux than windows</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>8bitcerberus</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Pretty much all of them, GIMP, Krita, Inkscape, Scribus, Kdenlive, Blender, etc. Faster than Windows and way faster than OS X. Blender is probably the closest I've seen between the 3 OSes, but it's still measurably faster on Linux, the others though, there's no contest.</p></div>
    </div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>pooerh</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Yeah that's probably mostly because you have gtk libs in shared memory from either other open apps or even your entire DE, while GIMP is the only thing that needs them on Windows.</p></div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>one_is_the_loneliest</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>Don't Windows apps typically bundle their own deps, so you'll end up with several of the same dep in memory at a time?</p><p>However, libraries are typically pretty small, so I'm guessing it doesn't explain most of the difference, but it's certainly a contributor.</p></div>
        <div class='comment' style='display: block; margin-left: 80px'>
          <h3>pooerh</h3>
          <div class='by' style='font-style: italic;'></div>
          <div class='comment_content'><p>Whole UI libraries like gtk or Qt are pretty big actually, which is why they usually start slower on Windows than on Linux. Most Windows apps are written using either Win32 API or .net and these aren't bundled with apps, these usually start up pretty quickly.</p></div>
          <div class='comment' style='display: block; margin-left: 80px'>
            <h3>one_is_the_loneliest</h3>
            <div class='by' style='font-style: italic;'></div>
            <div class='comment_content'><p>Huh, I just figured that Windows apps would bundle their UI libs, but I guess that makes sense.</p></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>Swiftpaw22</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>Now, maybe they're not comparing fresh load to fresh load, but it wouldn't at all surprise me if it was true when comparing second+ load to second+ load as Linux has always shined very brightly when it came to caching games to RAM better than Windows. Secondary loads in most of the games I've seen load faster than on Windows.</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>pdp10</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Both the Linux storage I/O subsystem and the Linux filesystems themselves are known to be faster than Windows NTFS.</p><p>On the majority of modern games you don't see much of a difference, though, because the assets are packed into ordered binary files and then memory-mapped. In a way, the devs are working around something of a Windows weakness by doing that (though it's a performance win in general).</p><p>That's a common pattern -- developers architect the game to work around Windows weaknesses, and of course don't take advantage of the strengths on Linux/POSIX. In principle someone could go the other direction, and design for some of POSIX/Linux strengths; I've pondered the topic in posts here before but neither I nor anyone else has a coherent thesis on the topic yet.</p></div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>s0v3r1gn</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>The largest differences between EXT4 and NTFS that affects the I/O are the maximum cluster size and the “allocate-on-flush” method of allocating disk space.</p><p>The latest version of NTFS substantially increased the maximum cluster size and both the cluster size and the “allocate-on-flush” method are substantially less impactful on loading times on SSDs than they were on HDDs.</p><p>The largest difference in load times is probably caused by Windows memory management being far more zealous in purging garbage.</p></div>
        <div class='comment' style='display: block; margin-left: 80px'>
          <h3>piotrj3</h3>
          <div class='by' style='font-style: italic;'></div>
          <div class='comment_content'><p>This. On loading screen there is a lot of memory allocation and thread creation and so on and this is where linux is faster.</p></div>
        </div>
      </div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>Tom2Die</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>I've pondered the topic in posts here before but neither I nor anyone else has a coherent thesis on the topic yet.</p><p>I have faith that some day when you mature a bit more and become a PDP-11 you'll find the answers you seek.</p></div>
        <div class='comment' style='display: block; margin-left: 80px'>
          <h3>pdp10</h3>
          <div class='by' style='font-style: italic;'></div>
          <div class='comment_content'><p>And drop 20 bits from the word?</p></div>
          <div class='comment' style='display: block; margin-left: 80px'>
            <h3>Tom2Die</h3>
            <div class='by' style='font-style: italic;'></div>
            <div class='comment_content'><p>Wait, did they really? There must be a good reason, but I'm not too familiar with older tech.</p></div>
            <div class='comment' style='display: block; margin-left: 80px'>
              <h3>pdp10</h3>
              <div class='by' style='font-style: italic;'></div>
              <div class='comment_content'><p>The short version is that the PDP-10 is a very large 36-bit mainframe, and the PDP-11 is a 16-bit minicomputer that only took up a couple of racks originally -- a small fraction of the physical size. Even thought the model lines sound similar, they were very different product lines that were used for different purposes.</p><p>PDP-11s were the second model of computer to run Unix, and the place where most of Unix was invented. Only PDP-11s ran Unix until the late 1970s. Later, the most popular hardware to run Unix were VAXes, the 32-bit replacement for the venerable PDP-11s. DEC didn't really like it when customers bought their hardware but not their operating systems. Unix never ran on 36-bit machines. The big tens are rather unique in history.</p><p>However, the command-line syntax was quite similar on all of the DEC operating systems. Kildall used much of it in CP/M, and through that path, much of it went into DOS. DOS also inherited a bit of syntax from Xenix, which Microsoft had licensed a year or two prior to the IBM PC project.</p><p>Microsoft wanted to license Xenix to the different OEM computer manufacturers, and sell their apps for Xenix in addition to other operating systems, prior to the IBM PC opportunity falling into their laps. Microsoft did continue to make apps for other operating systems, most fervently MacOS, the home of Excel. Word was on Xenix first, and only much later came to the Mac and to the fledgling Windows.</p><p>Unix was originally invented for gaming and word processing, and Microsoft Word was originally made for Unix. History is stranger than you think.</p></div>
              <div class='comment' style='display: block; margin-left: 80px'>
                <h3>Tom2Die</h3>
                <div class='by' style='font-style: italic;'></div>
                <div class='comment_content'><p>The odd thing is that at one point in time I knew most of that, but it's the sort of trivia that if one doesn't engage with it it just vanishes from memory without notice.</p></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>ihjyuhgyhhg</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>Can confirm, witcher 3 and GTA sa loading on Linux is unbelievably faster than windows.</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>sixsupersonic</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Which is interesting considering those are windows games running in wine.</p></div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>ihjyuhgyhhg</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>Exactly and also w3 runs at 24fps compared to 29fps on windows. It was already amazing that game was working but this. Linux is future.</p></div>
      </div>
    </div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>airspeedmph</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>Not surprised, I have an old X-Plane benchmark where you can see also a striking difference in loading times (link at the respective test):</p><p>[1]https://youtu.be/M5ygXe9fWR4?t=12</p><p>I also see very long loading times on Windows for Arma3, Rust and a couple of others, so yeah, not surprised.</p><p>Edit TL;DW: X-Plane scenery and all loading in 00:56 min for SteamOS and 02:33 min for Windows.</p><p></p><p></p><p></p><p></p><p>[1] https://youtu.be/M5ygXe9fWR4?t=12</p></div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>Leopard1907</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>Well , that is a known perk of Linux</p><p>Once upon a time PenguinRecordings ( a Youtube channel ) was doing Linux game benchmarks.</p><p>He was always comparing load times on Linux vs Windows. Let it be Feral ports or something else , always Linux was faster one.</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>Two-Tone-</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>PenguinRecordings</p><p>I miss his benchmarks. :c</p></div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>Leopard1907</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>Yeah , i also miss.</p><p>His enthusiasm was really something else.</p><p>Right after doing Doom 2016 via Wine video , he gone mute.</p></div>
        <div class='comment' style='display: block; margin-left: 80px'>
          <h3>Swiftpaw22</h3>
          <div class='by' style='font-style: italic;'></div>
          <div class='comment_content'><p>Yep, sad, wonder what happened to him. Sounds like it may have been a side project while he was in school.</p></div>
          <div class='comment' style='display: block; margin-left: 80px'>
            <h3>Leopard1907</h3>
            <div class='by' style='font-style: italic;'></div>
            <div class='comment_content'><p>I hope he is enjoying SteamPlay like we do and living his life. :)</p></div>
            <div class='comment' style='display: block; margin-left: 80px'>
              <h3>Swiftpaw22</h3>
              <div class='by' style='font-style: italic;'></div>
              <div class='comment_content'><p>That doesn't mean we don't still need Linux game benchmarking. At least we have Phoronix and a few other youtube channels that do it somewhat.</p></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>mykro76</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>ITT - many apps and games do load faster on Linux.</p><p>This seems like something worth promoting to game devs as a benefit of developing on Linux. How much time must they spend staring at their own loading screen when tweaking and testing their game?</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>Swiftpaw22</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Good point!</p></div>
    </div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>ThenewLore</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>If it only would load the actual missions and not crash on the mission title...</p></div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>Atlas__risen</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>I noticed a big improvement in FFXV on Linux vs Windows as well.</p></div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>Offensive_joke_lord</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>My windows-using friend is always so impressed by how fast my games load. Well, it's happened twice, once with invisible inc and once with crusader kings II. I'm pretty sure his computer is better spec-wise as well</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>Swiftpaw22</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>On the first load, or subsequent loads, or all loads? Because Linux generally seems faster to load all games on subsequent loads at least, but also sometimes on first loads as well.</p></div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>Offensive_joke_lord</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>Both first loads actually, we only play, and have only played, those games together at my house</p><p>P.S. (yes, they're both singleplayer haha, we have fun playing singleplayer games and exchanging the controls.)</p><p>P.P.S. (we even divide controls, one time we had 3 people and we played the binding of isaac where one person moved, one attacked, and the other used items)</p></div>
        <div class='comment' style='display: block; margin-left: 80px'>
          <h3>Swiftpaw22</h3>
          <div class='by' style='font-style: italic;'></div>
          <div class='comment_content'><p>Cool, and sounds fun, playing together is always best! :3</p></div>
        </div>
      </div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>Greydmiyu</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>For me it is both. But that is because when I dumped Win10 off my gaming rig I purposely went the LVM route so I could clear off my largely unused SSD and make it a [1]cache for my HDD.</p><p>The problem I had before was that I would get a game, put it on the SSD for fast loading, then move on to another game and not swap the games out from the SSD. With LVM using the SSD as a cache anything I use often gets loaded onto the SSD. If it ever fills up stuff I've stopped using gets removed from SSD. No need for me to manage it manually.</p><p>Recently it has been Warframe and World of Warcraft that take up the bulk of my time. From boot to shutdown about 80-90% of my file-system calls hit the SSD. I have a 120Gb cache for a 2Tb drive and so far it's only about 60% utilized.</p><p>A few months of that without touching the Win10 drive I still have sitting there and I can drop the cache, add the 1Tb of space to my volume, expand my file system, put the cache back on and have a 3Tb seamless logical volume to play with. :)</p><p></p><p></p><p></p><p></p><p>[1] https://rwmj.wordpress.com/2014/05/22/using-lvms-new-cache-feature/</p></div>
        <div class='comment' style='display: block; margin-left: 80px'>
          <h3>Swiftpaw22</h3>
          <div class='by' style='font-style: italic;'></div>
          <div class='comment_content'><p>The problem I had before was that I would get a game, put it on the SSD for fast loading, then move on to another game and not swap the games out from the SSD. With LVM using the SSD as a cache anything I use often gets loaded onto the SSD. If it ever fills up stuff I've stopped using gets removed from SSD. No need for me to manage it manually.</p><p>I was about to ask wtf the point of this would be, lol, but that clears things up, thanks!</p><p>I guess since RAM is more expensive than a SSD, having the SSD cache things makes sense as long as RAM is still used first and foremost for file caching.</p><p></p><p>have a 3Tb seamless logical volume to play with</p><p>As long as you have a backup somewhere of all your important stuff! Since HDDs are big and cheap I prefer just using one big one. Hell, there are 10TB ones out now and higher.</p></div>
          <div class='comment' style='display: block; margin-left: 80px'>
            <h3>Greydmiyu</h3>
            <div class='by' style='font-style: italic;'></div>
            <div class='comment_content'><p>As long as you have a backup somewhere of all your important stuff! Since HDDs are big and cheap I prefer just using one big one. Hell, there are 10TB ones out now and higher.</p><p>These are the two largest drives I have. I used to have Win10 on the 1TB and used the 2TB to record gaming videos. When I wanted to switch I realized I could just drop Linux on the 2TB drive and have the 1TB Win10 to fall back to if my litmus test for Linux wasn't passed this time around (Warframe & WoW being playable).</p><p>Being able to merge the 1TB into the file system is just an added bonus for when I'm ready to get rid of the safety blanket. Been a month and I think I've thought about booting to that drive... once? But the desire to try Destiny 2 again passed. :)</p></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>yoshi314</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>too bad this beta won't load its own saves. i am reluctant to replay game from scratch again, because neither autosaves nor my manual saves will load.</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>Dakkaface</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Haven't have that issue myself, it's been working for me aside from some graphical glitches. Hopefully they get it working for everyone soon.</p></div>
    </div>
  </div>
  <div class='comment' style='display: block; margin-left: 80px'>
    <h3>MJBrune</h3>
    <div class='by' style='font-style: italic;'></div>
    <div class='comment_content'><p>I will say while I do think there is some sort of faster load on linux, one user randomly staying on a forum shouldn't really be big news and certainly shouldn't be held as true to any real means.</p></div>
    <div class='comment' style='display: block; margin-left: 80px'>
      <h3>Swiftpaw22</h3>
      <div class='by' style='font-style: italic;'></div>
      <div class='comment_content'><p>Linux loading times being faster for many games, especially during 2nd+ loads, is 100% confirmed. As for this particular game, other users have confirmed the same thing within this very post.</p><p>So, while wanting evidence is amiable, we already have it.</p></div>
      <div class='comment' style='display: block; margin-left: 80px'>
        <h3>MJBrune</h3>
        <div class='by' style='font-style: italic;'></div>
        <div class='comment_content'><p>I mean sure. And there are plenty of benchmarks to show it. I just think "10 times" is a bit excessive and random. I dunno. Am not trying to crap on your joke either.</p></div>
        <div class='comment' style='display: block; margin-left: 80px'>
          <h3>Swiftpaw22</h3>
          <div class='by' style='font-style: italic;'></div>
          <div class='comment_content'><p>I don't know how accurate "10" is hehe, but some of the benchmark videos show huge differences, so I totally get someone say "10 times faster". But we don't have to be overly anal, they just mean it's "much faster" I'm sure, and that much is confirmed.</p><p>Nothing against anal. :3</p></div>
        </div>
      </div>
    </div>
  </div>
</div>
</body>
